apl-updater/src/main/java/com/apollocurrency/aplwallet/apl/updater/export/StartUpdateEventObserver.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.updater.export;
+
+import static com.apollocurrency.aplwallet.apl.core.app.service.SecureStorageService.SECURE_STORE_KEY;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Properties;
+
+import com.apollocurrency.aplwallet.apl.core.db.model.OptionDAO;
+import com.apollocurrency.aplwallet.apl.crypto.Convert;
+import com.apollocurrency.aplwallet.apl.crypto.Crypto;
+import com.apollocurrency.aplwallet.apl.updater.export.event.StartUpdateEvent;
+import com.apollocurrency.aplwallet.apl.updater.export.event.StartUpdateEventData;
+import com.apollocurrency.aplwallet.apl.updater.export.event.StartUpdateEventType;
+import com.apollocurrency.aplwallet.apl.util.env.dirprovider.ConfigDirProvider;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Class process sent event afer update is processed and not run script.
+ * It stores some properties into config folder for later use.
+ */
+@Slf4j
+@Singleton
+public class StartUpdateEventObserver {
+
+    public static final String EXPORTED_DATA_FILE_NAME = "exported-data.properties";
+
+    private PropertiesHolder propertiesHolder;
+    private OptionDAO optionDAO;
+    private ConfigDirProvider configDirProvider;
+
+    @Inject
+    public StartUpdateEventObserver(PropertiesHolder propertiesHolder, OptionDAO optionDAO, ConfigDirProvider configDirProvider) {
+        this.propertiesHolder = Objects.requireNonNull(propertiesHolder, "propertiesHolder is NULL");
+        this.optionDAO = Objects.requireNonNull(optionDAO, "optionDAO is NULL");
+        this.configDirProvider = Objects.requireNonNull(configDirProvider, "configDirProvider is NULL");
+    }
+
+    public void onStartUpdateBefore(
+        @Observes @StartUpdateEvent(StartUpdateEventType.BEFORE_SCRIPT) StartUpdateEventData startUpdateEventData) {
+        log.debug("onStartUpdateBefore...");
+        // read some properties
+        String nameKey = new String(Base64.getDecoder().decode("YXBsLmFkbWluUGFzc3dvcmQ="));
```

Comment:
YXBsLmFkbWluUGFzc3dvcmQ= looks like better move it to constants and name properly

Reply:
No, we don't need, let it be as is.